### PR TITLE
Purge region declarations

### PR DESCRIPTION
**Pull request by Nihlus**
This PR removes all region declarations from the codebase. These declarations are redundant in any modern development environment, and are severely overused (some of them wrapping single methods or properties).

Their removal reduces source size, line count, and code clutter.

### CODE REVIEW

File: src/Generator.Converter/Options.cs [hunk 1]
```diff
@@ -159,43 +159,32 @@ internal OptionValueCollection (OptionContext c)
...
 		public void CopyTo (string[] array, int arrayIndex) {values.CopyTo (array, arrayIndex);}
 		public bool Remove (string item)                    {return values.Remove (item);}
 		public int Count                                    {get {return values.Count;}}
 		public bool IsReadOnly                              {get {return false;}}
-		#endregion
 
-		#region IEnumerable
-		IEnumerator IEnumerable.GetEnumerator () {return values.GetEnumerator ();}
-		#endregion
+				IEnumerator IEnumerable.GetEnumerator () {return values.GetEnumerator ();}
```
Comment (Frassle) [inline src/Generator.Converter/Options.cs]: Oddly indented

File: src/Generator.Converter/Options.cs [hunk 2]
```diff
@@ -159,43 +159,32 @@ internal OptionValueCollection (OptionContext c)
...
 
-		#region IEnumerable<T>
-		public IEnumerator<string> GetEnumerator () {return values.GetEnumerator ();}
-		#endregion
+				public IEnumerator<string> GetEnumerator () {return values.GetEnumerator ();}
```
Comment (Frassle) [inline src/Generator.Converter/Options.cs]: Also oddly indented

File: src/Generator.Converter/Options.cs [hunk 3]
```diff
@@ -159,43 +159,32 @@ internal OptionValueCollection (OptionContext c)
...
 
-		#region IList
-		int IList.Add (object value)                {return (values as IList).Add (value);}
+				int IList.Add (object value)                {return (values as IList).Add (value);}
```
Comment (Frassle) [inline src/Generator.Converter/Options.cs]: And again

File: src/Generator.Converter/Options.cs [hunk 4]
```diff
@@ -159,43 +159,32 @@ internal OptionValueCollection (OptionContext c)
...
 		void IList.Insert (int index, object value) {(values as IList).Insert (index, value);}
 		void IList.Remove (object value)            {(values as IList).Remove (value);}
 		void IList.RemoveAt (int index)             {(values as IList).RemoveAt (index);}
 		bool IList.IsFixedSize                      {get {return false;}}
 		object IList.this [int index]               {get {return this [index];} set {(values as IList)[index] = value;}}
-		#endregion
 
-		#region IList<T>
-		public int IndexOf (string item)            {return values.IndexOf (item);}
+				public int IndexOf (string item)            {return values.IndexOf (item);}
```
Comment (Frassle) [inline src/Generator.Converter/Options.cs]: And again :)

File: src/Generator.Converter/Options.cs (hunk 1 above)
Comment (Nihlus) [inline src/Generator.Converter/Options.cs]: It appears it reindented with tabs, for some strange reason. I'll correct it.

Comment (Nihlus) [inline src/Generator.Converter/Options.cs]: @Frassle In fact, looking at the source, this is a prevalent problem in a lot of source files. Perhaps it's better to perform a second cosmetic pass after this PR and replace the tabs with spaces everywhere.

Comment (Frassle) [inline src/Generator.Converter/Options.cs]: I'd fix these while your there but yes I'd be happy with a no-tabs policy. Is there someway we can get the CI tools to give an error if pull requests come in with tabs? Not worth doing a bulk replace till we can be sure it won't just creep in again.

**Comment by Nihlus** [comment]
@Frassle The malformed indentation has been corrected. We could implement a set of StyleCop rules, but that's a fairly hefty undertaking, especially with the state of the style being as it is already. It would almost certainly be easier to perform a set of normalization passes and then implement StyleCop rules to make sure they don't come back. In doing so, however, we should adopt a complete set of contribution guidelines and a proper style guide as well.

**Comment by Frassle** [comment]
LGTM

**Pull request merged**